Bearded.Utilities/Noise/PerlinNoise.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Linq;
+using Bearded.Utilities.Geometry;
+using OpenTK.Mathematics;
+
+namespace Bearded.Utilities.Noise
+{
+    public static class PerlinNoise
+    {
+        // Use uniformly distributed vectors to force better gradient distribution.
+        private const int numVectorSamples = 12;
+        private static readonly Vector2d[] vectorSamples = vectorSamples = Enumerable.Range(0, numVectorSamples)
+            .Select(i => Direction2.FromRadians(MathConstants.TwoPi * i / numVectorSamples).Vector)
+            .Select(vector => new Vector2d(vector.X, vector.Y))
+            .ToArray();
+
+        public static INoiseMap Generate(int numCellsX, int numCellsY, int? seed)
+        {
+            if (numCellsX <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(numCellsX));
+            }
+            if (numCellsY <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(numCellsY));
+            }
+
+            var gradientArray = NoiseUtils.GenerateRandomArray(
+                // We generate the corners, but to make the noise map wrap along both dimensions, we reuse the same
+                // values as the left and top boundaries, so we don't have to generate the right and bottom boundaries.
```

Comment:
Similar to above, why do we wannt it to wrap? Should that be the default? Should that be configurable?

Reply:
> Similar to above, why do we wannt it to wrap?

By default, yes.

> Should that be the default?

Yes, this sounds the most expected and is in line with OpenGL defaults for traditional textures, and this doesn't immediately break cases that don't expect it.

> Should that be configurable?

Yes, to be added in a future PR.